src/DBConnector/Connector.java

```diff
@@ -75,49 +113,56 @@ public void printAllUsers() {
    */
   public boolean addAccount(String username, String password) {
 
-    /* string constructed for database */
-    String sqlAdd = "insert into "+tableName + "(username, password)" + "values(" + "'" + username
-        + "'" + "," + "'" + password + "'" + ")";
-    String sqlCount = "select count(*) from "+tableName;
-    String sqlSelect = "select username from "+tableName+" where username = " + "'" + username + "'";
-
+    ResultSet resultSize = null, resultUser= null;
     try {
       // acquire the table size from the database, and initialize the table size
-      ResultSet resultSize = myStmt.executeQuery(sqlCount);
+      resultSize = selectCount.executeQuery();
       int size = 0;
-
       // move the cursor one row forward, if there is a row, return true; and reassign value to size
       if (resultSize.next()) {
         size = resultSize.getInt(1);
         // move the cursor all the way back to the starting point
         resultSize.beforeFirst();
       }
-
       // if the table is empty, we directly insert the new account
       if (size == 0) {
-        myStmt.executeUpdate(sqlAdd);
+        add.setString(1,username);
+        add.setString(2,password);
+        add.executeUpdate();
+        resultSize.close();
         return true;
       }
-
       /*
        * if there exists at least one element, check the existence; we first get the array of row
        * that has required username
        */
-      ResultSet result = myStmt.executeQuery(sqlSelect);
-
+      selectUsername.setString(1,username);
+      resultUser = selectUsername.executeQuery();
       // check if there is any row that have the required username
-      if (result.next()) {
+      if (resultUser.next()) {
         // move the cursor all the way back to the starting point
-        result.beforeFirst();
+        resultUser.beforeFirst();
+        resultSize.close();
+        resultUser.close();
         return false;
       }
 
       // the newly created username does not exist in the database, then add it to the database
-      myStmt.executeUpdate(sqlAdd);
+      add.setString(1,username);
+      add.setString(2,password);
```

Comment:
If you're feeling industrious, this chain of calls makes the case for a Builder for this statement — using `setString(` and an index value is error prone.